app/views/share/_share.html.erb

```diff
@@ -4,7 +4,8 @@
   
   <% # HTML share code %>
```

Comment:
I believe the bookmark cache key isn't guaranteed to change when the bookmarkable does, so I'd suggest using a combined key for the bookmark cache, including both `bookmark.cache_key` and `bookmarkable.cache_key`. (Possibly with `Work.work_blurb_tag_cache(work.id)` thrown in there somewhere for bookmarked works.)

Reply:
Before we add caching, it would also help to [move the sharing options to a separate action](https://otwarchive.atlassian.net/browse/AO3-5907).

Reply:
Instead of dealing with cache invalidation, we could significantly improve load time by [having the sharing options loaded on demand](https://otwarchive.atlassian.net/browse/AO3-5987), like we did with works (#3815).